docs/content/guides/developer/app-examples/escrow.mdx

```diff
@@ -3,4 +3,160 @@ title: Escrow
+- `EMismatchedExchangeObject`: The `exchange_for` fields of the two escrowed objects do not match.
+
+```move
+const EMismatchedSenderRecipient: u64 = 0;
+const EMismatchedExchangeObject: u64 = 1;
+```
+
+#### Public Functions
+
+##### `create`
+
+This function is used to create a new escrow object. It takes four arguments: the object to be escrowed, the ID of the key that opens the lock on the object sender wants from the recipient, the intended recipient, and the transaction context.
+
+```move
+public fun create<T: key + store>(
+   escrowed: T,
+   exchange_key: ID,
+   recipient: address,
+   ctx: &mut TxContext
+) {
+   let escrow = Escrow {
+       id: object::new(ctx),
+       sender: tx_context::sender(ctx),
+       recipient,
+       exchange_key,
+       escrowed,
+   };
+
+   transfer::public_share_object(escrow);
+}
+```
+
+##### `swap`
+
+This function is used to perform the swap operation. It takes four arguments: the escrow object, the key, the locked object, and the transaction context.
+
+```move
+public fun swap<T: key + store, U: key + store>(
+   escrow: Escrow<T>,
+   key: Key,
+   locked: Locked<U>,
+   ctx: &TxContext,
+): T {
+   let Escrow {
+       id,
+       sender,
+       recipient,
+       exchange_key,
+       escrowed,
+   } = escrow;
+
+   assert!(recipient == tx_context::sender(ctx), EMismatchedSenderRecipient);
+   assert!(exchange_key == object::id(&key), EMismatchedExchangeObject);
+
+   // Do the actual swap
+   transfer::public_transfer(lock::unlock(locked, key), sender);
+   object::delete(id);
+
+   escrowed
+}
+```
+
+##### `return_to_sender`
+
+This function is used to cancel the escrow and return the escrowed item to the sender. It takes two arguments: the escrow object and the transaction context.
+
+```move
+public fun return_to_sender<T: key + store>(
+   escrow: Escrow<T>,
+   ctx: &TxContext
+): T {
+   let Escrow {
+       id,
+       sender,
+       recipient: _,
+       exchange_key: _,
+       escrowed,
+   } = escrow;
+
+   assert!(sender == tx_context::sender(ctx), EMismatchedSenderRecipient);
+   object::delete(id);
+   escrowed
+}
+```
+
+#### Tests
+
+The code includes several tests to ensure the correct functioning of the atomic swap process. These tests cover successful swaps, mismatches in sender or recipient, mismatches in the exchange object, tampering with the object, and returning the object to the sender.
+
+In conclusion, this code provides a robust and secure way to perform atomic swaps of objects in a decentralized system, without the need for a trusted third party. It uses shared objects and a series of checks to ensure that the swap only occurs if all conditions are met.
+
+#### Deployment
+
+Now that we have written the smart contract, let’s deploy it. First ensure the contract compiles successfully using `sui move build`. Once that is confirmed, you can deploy the project using the command `sui client publish --gas-budget 10000000`.
```

Comment:
cc @manolisliolios , I think with the introduction of the backend, the deployment process will change.